### PR TITLE
Modeler allocation input

### PR DESCRIPTION
**Pull request by SouthEndMusic**
Fixes https://github.com/Deltares/Ribasim/issues/656.

### CODE REVIEW

**Comment by SouthEndMusic** [comment]
For some reason I now get an `SQLiteException("out of memory")`, M'W'E:

```julia
using Ribasim
using SQLite

toml_path = "path/to/generated_testmodels/subnetwork/ribasim.toml"
cfg = Ribasim.Config(toml_path)
gpkg_path = Ribasim.input_path(cfg, cfg.database)
db = SQLite.DB(gpkg_path)
p = Ribasim.Parameters(db, cfg)
close(db)

Ribasim.generate_allocation_models!(p, db, cfg)
```

**Comment by SouthEndMusic** [comment]
@reviewer: I suggest adding more tests after https://github.com/Deltares/Ribasim/issues/659.

File: core/src/bmi.jl [hunk 1]
```diff
@@ -223,6 +229,13 @@ function create_callbacks(
     tabulated_rating_curve_cb = PresetTimeCallback(tstops, update_tabulated_rating_curve!)
     push!(callbacks, tabulated_rating_curve_cb)
 
+    if config.allocation.use_allocation
+        t_end = seconds_since(config.endtime, config.starttime)
+        tstops = 1:(config.allocation.timestep):t_end
```
Comment (visr) [inline core/src/bmi.jl:234]: Do you intentionally start after 1 second rather than at the beginning?

File: core/src/bmi.jl [hunk 2]
```diff
@@ -223,6 +229,13 @@ function create_callbacks(
...
+        allocation_cb = PresetTimeCallback(tstops, update_allocation!)
```
Comment (visr) [inline core/src/bmi.jl:235]: For a fixed interval the PeriodicCallback is a better match: https://docs.sciml.ai/DiffEqCallbacks/stable/timed_callbacks/

**Review by visr — COMMENTED** [review]
Nice work! Left a few comments.

File: core/src/bmi.jl (hunk 1 above)
Comment (SouthEndMusic) [reply core/src/bmi.jl:234]: No, this is probably due to me not being completely used to 1-based indexing (although I know this doesn't actually have anything to do with that) ^^

**Pull request merged**